stingray/powerspectrum.py

```diff
@@ -417,3 +414,196 @@ def _make_segment_spectrum(self, lc, segment_size):
+        """
+        Parameters
+        ----------
+        lc : lightcurve.Lightcurve object
+            The time series of which the Dynamical powerspectrum is
+            to be calculated.
+
+        segment_size : float, default 1
+             Length of the segment of light curve, default value is 1 second.
+
+        norm: {"leahy" | "frac" | "abs" | "none" }, optional, default "frac"
+            The normaliation of the periodogram to be used. Options are
+            "leahy", "frac", "abs" and "none", default is "frac".
+
+        Other Parameters
+        ----------------
+        gti: 2-d float array
+            [[gti0_0, gti0_1], [gti1_0, gti1_1], ...] -- Good Time intervals.
+            This choice overrides the GTIs in the single light curves. Use with
+            care!
+
+        Attributes
+        ----------
+        segment_size: float
+            The size of each segment to average. Note that if the total
+            duration of each Lightcurve object in lc is not an integer multiple
+            of the segment_size, then any fraction left-over at the end of the
+            time series will be lost.
+
+        dyn_ps : np.ndarray
+            The matrix of normalized squared absolute values of Fourier
+            amplitudes. The axis are given by the `freq`
+            and `time` attirbutes
+
+        norm: {"leahy" | "frac" | "abs" | "none"}
+            the normalization of the periodogram
+
+        freq: numpy.ndarray
+            The array of mid-bin frequencies that the Fourier transform samples
+
+        df: float
+            The frequency resolution
+
+        dt: float
+            The time resolution
+        """
+        if segment_size < 2 * lc.dt:
+            raise ValueError("Length of the segment is too short to form a "
+                             "light curve!")
+        elif segment_size > lc.tseg:
+            raise ValueError("Length of the segment is too long to create "
+                             "any segments of the light curve!")
+        self.segment_size = segment_size
+        self.norm = norm
+        self.gti = gti
+        self._make_matrix(lc)
+
+    def _make_matrix(self, lc):
+        """Create the matrix with freq and time columns."""
+        ps_all, _ = AveragedPowerspectrum._make_segment_spectrum(
+            self, lc, self.segment_size)
+        self.dyn_ps = np.array([ps.power for ps in ps_all]).T
+
+        self.freq = ps_all[0].freq
+        self.time = np.arange(lc.time[0] - 0.5*lc.dt + 0.5*self.segment_size,
+                              lc.time[-1] + 0.5*lc.dt, self.segment_size)
+
+        # Assign zero resolution if only one value
+        if len(self.time) > 1:
+            self.dt = self.time[1] - self.time[0]
+        else:
+            self.dt = 0
+
+        # Assign Zero freq. resolution if only one value
+        if len(self.freq) > 1:
+            self.df = self.freq[1] - self.freq[0]
+        else:
+            self.df = 0
```

Comment:
Is this something that can happen? Anyway, similarly to above, I think that df should be the frequency resolution, or 1/length.